### ISSUE TITLE
Move output mappings under its own sub tree eg ${{steps.ws.outputs.myvar}}

### ISSUE THREAD

**Issue by brendanheywood** [catalyst/moodle-tool_dataflows#525]
* [ ] move step level vars under 'vars' 
* [ ] rename flow level config to 'vars' (do this deeply, var names and method names)
* [ ] add flow level config with enabled as the real config ie dataflow.config.enabled

**Comment by keevan** [comment]
I'd go with `.outputs.*` (from output) since there could be multiple fields, and matches `steps.*`

**Comment by brendanheywood** [comment]
Or what about ${{steps.ws.vars.myvar}} because then it is consistent with both global vars. Flow also have vars but they are called 'config' and I think we should make them vars too, config is for things which have well defined 

${{global.vars.globallevelvar}}
${{steps.ws.vars.steplevelvar}}
${{dataflow.vars.flowlevelvar}}

flow level config should be things like if it is enabled and allows concurrency:

${{dataflow.config.enabled}}
${{dataflow.config.concurrencyenabled}}

**Comment by keevan** [comment]
Originally, I thought outputs made sense because you'd ideally set it based on some result of the step. The terminology also suggests something is coming out from the step. 

I'm still leaning towards using outputs instead of vars, because I would like to not encourage the setting of vars from any step that "can" as opposed to the step related to the step's result, and vars is a bit generic (much like using data for the iterator value that gets passed along a flow).

In terms of vars vs config, sure, global vars is technically a collection of predefined variables. But those variables aren't typically fitting their root meaning ("not consistent or having a fixed pattern; liable to change."), but rather, act moreso as configuration since it's fixed. In that sense, I would lean towards making global.vars `global.config` instead (probably even changing it from global.cfg if it's set as such currently), and adding to it. 

If I may be bold and suggest that it might make even more sense if we change the current `steps.config.somefield` from `config` to `input`, because essentially, that's what they are. They are inputs required or expected in some shape or form, for that step, and they produce some output (which can be remapped by the user should they chose to).

tl;dr: 
- `config` over `vars` (at least for globals) because config is set and unchanged (e.g. before being run), vars implies the fact it's varying and can change.
- `outputs` instead of config/vars because unlike config, outputs should be derived from a step's result or output, and using this terminology should encourage that.
- `config` to `inputs` for current step type config fields because they are required/expected for the step type to run and produce said output.,

**Comment by jaypha** [comment]
So what's it going to be? 'config' or 'vars'.

**Comment by jaypha** [comment]
I have noticed that step form settings are stored in two groups. The settings that are common to all steps are set at the step level (e.g. `steps.somestep.name`), while the custom settings are stored under 'config' (e.g. `steps.somecurl.config.destination`). I'm thinking that they all should be in the same location, either under 'config' or at the step level.

What say you all?

**Comment by brendanheywood** [comment]
exec call to name it vars, lets do that first here and we'll come back to the config question

**Comment by jaypha** [comment]
Does this mean leave it alone? Or that everything is to be under vars?
